master/docs/manual/cfg-wwwhooks.rst

```diff
@@ -25,6 +25,37 @@ The :file:`post_build_request.py` script in :file:`master/contrib` allows for th
 Run :command:`post_build_request.py --help` for more information.
 The ``base`` dialect must be enabled for this to work.
 
+.. _Change-Hooks-Auth:
+
+Change Hooks Auth
++++++++++++++++++
+
+By default change hook URL is not protected.
+Some hooks implement their own authentication method.
+Other requires the generic method to be secured.
+
+To protect URL against unauthorized access you you may use ``change_hook_auth`` option.
+
+.. note::
+
+    This method uses ``HTTP BasicAuth``, it implies the use of SSL via :ref:`Reverse_Proxy_Config` in order to be fully secured.
+
+.. code-block:: python
+
+    from twisted.cred import strcred
+    c['www'] = dict(...,
+          change_hook_auth=[strcred.makeChecker("file:changehook.passwd")]))
+
+create a file ``changehook.passwd``:
+
+.. code-block:: none
+
+    user:password
```

Comment:
as suggested, please look at twisted documentation for the limitations. I dont think there are any